pkg/reconciler/v1alpha1/broker/broker.go

```diff
@@ -0,0 +1,442 @@
+	c, err := r.getChannel(ctx, b)
+	// If the resource doesn't exist, we'll create it
+	if k8serrors.IsNotFound(err) {
+		c = newChannel(b)
+		err = r.client.Create(ctx, c)
+		if err != nil {
+			return nil, err
+		}
+		return c, nil
+	} else if err != nil {
+		return nil, err
+	}
+
+	// TODO Determine if we want to update spec (maybe just args?).
+	// Update Channel if it has changed. Note that we need to both ignore the real Channel's
+	// subscribable section and if we need to update the real Channel, retain it.
+	//expected.Spec.Subscribable = c.Spec.Subscribable
+	//if !equality.Semantic.DeepDerivative(expected.Spec, c.Spec) {
+	//	c.Spec = expected.Spec
+	//	err = r.client.Update(ctx, c)
+	//	if err != nil {
+	//		return nil, err
+	//	}
+	//}
+	return c, nil
+}
+
+// getChannel returns the Channel object for Broker 'b' if exists, otherwise it returns an error.
+func (r *reconciler) getChannel(ctx context.Context, b *v1alpha1.Broker) (*v1alpha1.Channel, error) {
+	list := &v1alpha1.ChannelList{}
+	opts := &runtimeclient.ListOptions{
+		Namespace:     b.Namespace,
+		LabelSelector: labels.SelectorFromSet(ChannelLabels(b)),
+		// Set Raw because if we need to get more than one page, then we will put the continue token
+		// into opts.Raw.Continue.
+		Raw: &metav1.ListOptions{},
+	}
+
+	err := r.client.List(ctx, opts, list)
+	if err != nil {
+		return nil, err
+	}
+	for _, c := range list.Items {
+		if metav1.IsControlledBy(&c, b) {
+			return &c, nil
+		}
+	}
+
+	return nil, k8serrors.NewNotFound(schema.GroupResource{}, "")
+}
+
+// newChannel creates a new Channel for Broker 'b'.
+func newChannel(b *v1alpha1.Broker) *v1alpha1.Channel {
+	var spec v1alpha1.ChannelSpec
+	if b.Spec.ChannelTemplate != nil {
+		spec = *b.Spec.ChannelTemplate
+	}
+
+	return &v1alpha1.Channel{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    b.Namespace,
+			GenerateName: fmt.Sprintf("%s-broker-", b.Name),
+			Labels:       ChannelLabels(b),
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(b, schema.GroupVersionKind{
+					Group:   v1alpha1.SchemeGroupVersion.Group,
+					Version: v1alpha1.SchemeGroupVersion.Version,
+					Kind:    "Broker",
+				}),
+			},
+		},
+		Spec: spec,
+	}
+}
+
+func ChannelLabels(b *v1alpha1.Broker) map[string]string {
+	return map[string]string{
+		"eventing.knative.dev/broker":           b.Name,
+		"eventing.knative.dev/brokerEverything": "true",
+	}
+}
+
+// reconcileDeployment reconciles the K8s Deployment 'd'.
+func (r *reconciler) reconcileDeployment(ctx context.Context, d *v1.Deployment) (*v1.Deployment, error) {
+	name := types.NamespacedName{
+		Namespace: d.Namespace,
+		Name:      d.Name,
+	}
+	current := &v1.Deployment{}
+	err := r.client.Get(ctx, name, current)
+	if k8serrors.IsNotFound(err) {
+		err = r.client.Create(ctx, d)
+		if err != nil {
+			return nil, err
+		}
+		return d, nil
+	} else if err != nil {
```

Comment:
nit: no need for 'else'. 'If' is enough

Reply:
These feel 'linked' in my mind, so I'm OK with using the else. And it saves a line...